action/protocol/account/transfer.go

```diff
@@ -117,7 +117,7 @@ func (p *Protocol) handleTransfer(ctx context.Context, act action.Action, sm pro
 		return nil, errors.Wrap(err, "failed to update pending account changes to trie")
 	}
 	// check recipient
-	recipient, err := accountutil.LoadOrCreateAccount(sm, tsf.Recipient())
+	recipient, err := accountutil.LoadOrCreateAccount(sm, recipientAddr)
```

Comment:
`tsf.Recipient()` is the string format, so we need to check similar as `address.FromLegacy()`
instead of doing that, directly use the `recipientAddr`